src/olympia/blocklist/cron.py

```diff
@@ -119,28 +116,25 @@ def _upload_mlbf_to_remote_settings(*, force_base=False):
         len(mlbf.data.not_blocked_items),
     )
 
-    make_base_filter = (
-        force_base
-        or base_filter is None
-        or previous_filter is None
-        or mlbf.blocks_changed_since_previous(BlockType.BLOCKED, base_filter)
-        > BASE_REPLACE_THRESHOLD
-    )
+    # Until we are ready to enable soft blocking, it should not be possible
+    # to create a stash and a filter at the same iteration
+    if create_stash and len(base_filters_to_update) > 0:
+        raise Exception(
+            'Cannot upload stash and filter without implementing soft blocking'
+        )
```

Comment:
Do we have test coverage for this error?

Reply:
It's actually impossible to cover because we force skip soft block, and if hard block produces a filter, it cannot produce a stash.. so not really. I'd have to do some serious hacking. We never expect this error. I'll add a comment as such.

Reply:
This might be a case where we want to use `assert` since that sounds like an invariant?

Reply:
Got it.